lib/modules/versioning/poetry/transform.ts

```diff
@@ -63,12 +63,18 @@ export function poetry2semver(
 
   const parts = [releaseParts.map((num) => num.toString()).join('.')];
   if (pre !== null) {
+    // trim leading zeros from valid numbers
+    pre.number = parseInt(pre.number, 10).toString();
```

Comment:
🤔 isn't a regex replace faster that parsing a number and stringify again?

Reply:
done, but I needed to update your regex some since it didn't account for numbers other than zero

e.g. `001` would become `01` instead of `1`